pkg/cli/admin/inspectalerts/inspectalerts.go

```diff
@@ -0,0 +1,133 @@
+// Package inspectalerts provides access to in-cluster alerts.
+package inspectalerts
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/client-go/rest"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+	routev1 "github.com/openshift/api/route/v1"
+	routev1client "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+)
+
+// RouteGetter is a function that gets a Route.
+type RouteGetter func(ctx context.Context, namespace string, name string, opts metav1.GetOptions) (*routev1.Route, error)
+
+type Alert struct {
+}
+
+type options struct {
+	genericiooptions.IOStreams
+	RESTConfig *rest.Config
+	getRoute   RouteGetter
+}
+
+func newOptions(streams genericiooptions.IOStreams) *options {
+	return &options{
+		IOStreams: streams,
+	}
+}
+
+func New(f kcmdutil.Factory, streams genericiooptions.IOStreams) *cobra.Command {
+	o := newOptions(streams)
+	cmd := &cobra.Command{
+		Use:   "inspect-alerts",
+		Short: "Collect information about running alerts.",
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	return cmd
+}
+
+func (o *options) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	kcmdutil.RequireNoArguments(cmd, args)
+
+	cfg, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	o.RESTConfig = cfg
+
+	routeClient, err := routev1client.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	o.getRoute = func(ctx context.Context, namespace string, name string, opts metav1.GetOptions) (*routev1.Route, error) {
+		return routeClient.Routes(namespace).Get(ctx, name, opts)
+	}
+
+	return nil
+}
+
+func (o *options) Run(ctx context.Context) error {
+	alertBytes, err := GetWithBearer(ctx, o.getRoute, "openshift-monitoring", "alertmanager-main", "/api/v2/alerts", o.RESTConfig.BearerToken)
+	if err != nil {
+		return err
+	}
+
+	_, err = o.Out.Write(alertBytes)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetWithBearer gets a Route by namespace/name, contructs a URI using
+// status.ingress[].host and the path argument, and performs GETs on that
```

Comment:
Maybe https://github.com/openshift/library-go/blob/2ad786549f07f264339f63b12b71a0fe5e77e832/pkg/route/routeapihelpers/routeapihelpers.go#L18 could help here, it also checks the conditions ;)

Reply:
`oc` vendors library-go, but not yet that sub-package^, and I wanted to avoid mucking with `vendor/` and introduce thinking about iteration over `ingress[]` (`IngressURI` only returns the first match).  But yes, we could pivot to using that.  Or extend library-go to grow a callback-consuming function that supports iteration over multiiple matches, etc.  Thoughts for this pull?

Reply:
We can leave that for future iterations for sure.